### ISSUE TITLE
CCE thrown when compiling transaction code

### ISSUE THREAD

**Issue by rasika** [ballerina-platform/ballerina-lang#24833]
**Description:**
A CCE thrown when compiling a transaction related code.
```
java.lang.ClassCastException: io.ballerinalang.compiler.syntax.tree.MappingBindingPatternNode cannot be cast to io.ballerinalang.compiler.syntax.tree.CaptureBindingPatternNode
```

**Steps to reproduce:**
Consider following code snippet. When compiling below code, the ClassCastException is thrown. 
```ballerina
import ballerina/http;

function initiateNestedTransactionInRemote(string nestingMethod) returns string {
   http:Client remoteEp = new("http://localhost:8889");
    string s = "";
    transaction {
        string testString = "Hello World!!";
        testString.indexOf()
    } onretry {
        s += " onretry";
    } committed {
        s += " committed";
    } aborted {
        s += " aborted";
    }
    return s;
}
```
**Affected Versions:**
2.0.0-P3-SNAPSHOT

**Comment by SupunS** [comment]
This is not a valid syntax according to the new spec. Therefore, the new parser tries to recover and ends-up with a module level-var-decl with typed-binding-patterns.

However then it fails at the transformer due to https://github.com/ballerina-platform/ballerina-lang/issues/24878.

**Comment by rdulmina** [comment]
Fixed in #26949
